js/pro/timex.js

```diff
@@ -0,0 +1,622 @@
+        if (balances !== undefined) {
+            result = this.parseBalance (balances);
+        }
+        const balance = this.safeValue (payload, 'balance');
+        if (balance !== undefined) {
+            const currencyId = this.safeString (balance, 'currency');
+            const code = this.safeCurrencyCode (currencyId);
+            const account = this.account ();
+            account['total'] = this.safeString (balance, 'totalBalance');
+            account['used'] = this.safeString (balance, 'lockedBalance');
+            result[code] = account;
+        }
+        this.balance = this.safeBalance (result);
+        client.resolve (this.balance, 'balances');
+    }
+
+    async watchTicker (symbol, params = {}) {
+        /**
+         * @method
+         * @name timex#watchTicker
+         * @description watches a price ticker, a statistical calculation with the information calculated over the past 24 hours for a specific market
+         * @url https://docs.timex.io/websocket-api-list-patterns.html#ticker
+         * @param {string} symbol unified symbol of the market to fetch the ticker for
+         * @param {object} params extra parameters specific to the timex api endpoint
+         * @param {string} params.period period of ticker, I1, I5, I15, I30, H1, H2, H4, H6, H12, D1, W1
+         * @returns {object} a [ticker structure]{@link https://docs.ccxt.com/en/latest/manual.html#ticker-structure}
+         */
+        await this.loadMarkets ();
+        const market = this.market (symbol);
+        symbol = this.symbol (symbol);
+        const url = this.urls['api']['ws'];
+        const messageHash = 'ticker:' + symbol;
+        const period = this.safeString (params, 'period', 'D1');
+        params = this.omit (params, period);
+        const message = {
+            'type': 'SUBSCRIBE',
+            'requestId': this.requestId (),
+            'pattern': '/ticker/' + market['id'] + '/' + period,
+        };
+        const request = this.deepExtend (message, params);
+        return await this.watch (url, messageHash, request, messageHash, request);
+    }
+
+    async watchTrades (symbol, since = undefined, limit = undefined, params = {}) {
+        /**
+         * @method
+         * @name timex#watchTrades
+         * @description get the list of most recent trades for a particular symbol
+         * @url https://docs.timex.io/websocket-api-list-patterns.html#trade
+         * @param {string} symbol unified symbol of the market to fetch trades for
+         * @param {int|undefined} since timestamp in ms of the earliest trade to fetch
+         * @param {int|undefined} limit the maximum amount of trades to fetch
+         * @param {object} params extra parameters specific to the timex api endpoint
+         * @returns {[object]} a list of [trade structures]{@link https://docs.ccxt.com/en/latest/manual.html?#public-trades}
+         */
+        await this.loadMarkets ();
```

Comment:
Here the exchange returned raw trades but ccxt did not react so I'm assuming something's wrong with `handleMessage` or `handleTrades`, can you check it pls?
![image](https://user-images.githubusercontent.com/43336371/195124578-87ef95cf-bb48-41e8-bc92-657fc6263af0.png)